### PR TITLE
feat!: accept positional arguments when creating client

### PR DESCRIPTION
**Pull request by tdstein**
Modifies the Client initialization definition to optional accept the URL and API key as position arguments.

BREAKING CHANGE: Swaps the `api_key` and `url` argument positions in Client __init__.

### CODE REVIEW

**Comment by tdstein** [comment]
This change simplifies workflows in which CONNECT_API_KEY is provided via an environment variable, but the server URL is hardcoded. I believe this is the most common use case since most users do not have multiple Connect servers.

**Review by nealrichardson — APPROVED** [review]
Makes sense to me that server would be the first arg and api key after.

File: src/posit/connect/client.py [hunk 1]
```diff
@@ -47,11 +47,114 @@ class Client:
...
+    @overload
+    def __init__(self) -> None:
+        """Initialize a Client instance.
+
+        Creates a client instance using credentials read from the environment.
+
+        Environment Variables
+        ---------------------
+        CONNECT_API_KEY - The API key credential for client authentication.
+        CONNECT_SERVER - The Connect server URL.
```
[the remark this answers is not in the document]
Comment (tdstein) [reply src/posit/connect/client.py:59]: Seems logical. 
```suggestion
        Environment Variables
        ---------------------
        CONNECT_SERVER - The Connect server URL.
        CONNECT_API_KEY - The API key credential for client authentication.
```

File: src/posit/connect/client.py [hunk 2]
```diff
@@ -47,11 +47,114 @@ class Client:
...
+        """Initialize a Client instance."""
+        ...
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Initialize a Client instance.
+
+        Environment Variables
+        ---------------------
+        CONNECT_API_KEY - The API key credential for client authentication.
+        CONNECT_SERVER - The Connect server URL.
```
[the remark this answers is not in the document]
Comment (tdstein) [reply src/posit/connect/client.py:116]:
```suggestion
        Environment Variables
        ---------------------
        CONNECT_SERVER - The Connect server URL.
        CONNECT_API_KEY - The API key credential for client authentication.
```

**Pull request merged**